pyiceberg/table/__init__.py

```diff
@@ -1343,33 +1346,37 @@ class FileScanTask(ScanTask):
     delete_files: Set[DataFile]
     start: int
     length: int
+    residual: BooleanExpression
 
     def __init__(
         self,
         data_file: DataFile,
         delete_files: Optional[Set[DataFile]] = None,
         start: Optional[int] = None,
         length: Optional[int] = None,
+        residual: Optional[BooleanExpression] = None,
```

Comment:
I think we should make this one required. If there is no residual, it should default to `AlwaysTrue()`.